### PR TITLE
Adding GCI to node e2e.

### PR DESCRIPTION
**Pull request by vishh**
Depends on https://github.com/kubernetes/kubernetes/pull/29486
Adding the dev release as of now since stable and beta run docker v1.9.1
which is incompatible with kubelet.

### CODE REVIEW

**Comment by vishh** [comment]
cc @yinghan

**Comment by vishh** [comment]
cc @dchen1107

**Comment by Random-Liu** [comment]
```
Failure Finished Host tmp-node-e2e-6529eff2-gci-dev-53-8530-29-0 Test Suite
timeout: failed to run command './ginkgo': Permission denied

[command [ssh -i /home/jenkins/.ssh/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -o LogLevel=ERROR 104.198.210.124 -- sh -c 'cd /tmp/gcloud-e2e-940261792 && timeout -k 30s 2700s ./ginkgo --skip=FLAKY ./e2e_node.test -- --logtostderr --v 2 --build-services=false --stop-services=false --node-name=tmp-node-e2e-6529eff2-gci-dev-53-8530-29-0 --report-dir=/tmp/gcloud-e2e-940261792/results --junit-file-number=1 --cgroups-per-qos=true'] failed with error: exit status 126 and output:
timeout: failed to run command './ginkgo': Permission denied
, command [scp -i /home/jenkins/.ssh/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -o LogLevel=ERROR -r 104.198.210.124:/tmp/gcloud-e2e-940261792/results/ /var/lib/jenkins/workspace/node-pull-build-e2e-test/_artifacts/tmp-node-e2e-6529eff2-gci-dev-53-8530-29-0] failed with error: exit status 1 and output:
scp: /tmp/gcloud-e2e-940261792/results: No such file or directory
]
```

And #29486 got merged, please rebase this one. :)

**Comment by Random-Liu** [comment]
One thing needs mentioning that, in all our test image, we've moved the user `jenkins` to docker group when build the image (see [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/environment/setup_host.sh#L102)).

So we can set `SETUP_NODE=false` (see [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/jenkins/jenkins-ci.properties#L7)).

However, for GCI we didn't do that, we have to use `SETUP_NODE=true` which is different from the other images.

**Comment by vishh** [comment]
Hmm. Image metadata is not being passed down in Jenkins env. Fixing it now.

**Comment by vishh** [comment]
@k8s-bot test unit github issue #IGNORE

File: test/e2e_node/runner/run_e2e.go [hunk 1]
```diff
@@ -102,19 +113,28 @@ func main() {
...
 			glog.Fatalf("Could not read image config file provided: %v", err)
 		}
-		err = yaml.Unmarshal(imageConfigData, gceImages)
+		externalImageConfig := ImageConfig{Images: make(map[string]GCEImage)}
+		err = yaml.Unmarshal(imageConfigData, &externalImageConfig)
 		if err != nil {
 			glog.Fatalf("Could not parse image config file: %v", err)
 		}
+		for key, imageConfig := range externalImageConfig.Images {
+			gceimage := internalGCEImage{
```
Comment (Random-Liu) [inline test/e2e_node/runner/run_e2e.go]: s/gceimage/gceImage

File: test/e2e_node/runner/run_e2e.go [hunk 2]
```diff
@@ -124,23 +144,26 @@ func main() {
 			glog.Fatal("Must specify --image-project if you specify --images")
 		}
 		cliImages := strings.Split(*images, ",")
+
 		for _, img := range cliImages {
-			gceImages.Images[img] = GCEImage{
-				Image:   img,
-				Project: *imageProject,
+			gceimage := internalGCEImage{
```
Comment (Random-Liu) [inline test/e2e_node/runner/run_e2e.go]: s/gceimage/gceImage

File: test/e2e_node/runner/run_e2e.go [hunk 3]
```diff
@@ -124,23 +144,26 @@ func main() {
 			glog.Fatal("Must specify --image-project if you specify --images")
 		}
 		cliImages := strings.Split(*images, ",")
+
```
Comment (Random-Liu) [inline test/e2e_node/runner/run_e2e.go]: Unnecessary empty line.

**Comment by Random-Liu** [comment]
@k8s-bot test this issue https://github.com/kubernetes/kubernetes/issues/29451.

**Comment by Random-Liu** [comment]
LGTM with nits. And please squash. :)

Thanks a lot for making this work!

File: test/e2e_node/runner/run_e2e.go (hunk 1 above)
Comment (vishh) [inline test/e2e_node/runner/run_e2e.go]: Done

File: test/e2e_node/runner/run_e2e.go (hunk 2 above)
Comment (vishh) [inline test/e2e_node/runner/run_e2e.go]: Done

File: test/e2e_node/runner/run_e2e.go (hunk 3 above)
Comment (vishh) [inline test/e2e_node/runner/run_e2e.go]: Done

**Comment by vishh** [comment]
The commits are there on purpose.

**Comment by vishh** [comment]
Fixed comments. Applying LGTM label! Thanks for the review @Random-Liu !

**Comment by vishh** [comment]
Bumping priority for this PR since running node e2e on GCI is critical and has been missing all along.

**Pull request merged**